File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcStructuralActivity/IfcStructuralAction/IfcStructuralCurveAction/IfcStructuralCurveAction.cs
public class IfcStructuralCurveAction : IfcStructuralAction
{
    public IfcProjectedOrTrueLengthEnum ProjectedOrTrue { get; set; }
    public IfcStructuralCurveActivityTypeEnum PredefinedType { get; set; }

    // === EXTENDED BY ===
    // IfcStructuralLinearAction

    // === WHERE CLAUSES ===
    // HasObjectType : (PredefinedType <> IfcStructuralCurveActivityTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)
    // ProjectedIsGlobal : (NOT EXISTS(ProjectedOrTrue)) OR ((ProjectedOrTrue <> PROJECTED_LENGTH) OR (SELF\IfcStructuralActivity.GlobalOrLocal = GLOBAL_COORDS))
    // SuitablePredefinedType : PredefinedType <> IfcStructuralCurveActivityTypeEnum.EQUIDISTANT
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcStructuralActivity/IfcStructuralAction/IfcStructuralCurveAction/IfcStructuralCurveAction.vb
Public Class IfcStructuralCurveAction Inherits IfcStructuralAction
    Public ProjectedOrTrue As IfcProjectedOrTrueLengthEnum
    Public PredefinedType As IfcStructuralCurveActivityTypeEnum

    ' === EXTENDED BY ===
    ' IfcStructuralLinearAction

    ' === WHERE CLAUSES ===
    ' HasObjectType : (PredefinedType <> IfcStructuralCurveActivityTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)
    ' ProjectedIsGlobal : (NOT EXISTS(ProjectedOrTrue)) OR ((ProjectedOrTrue <> PROJECTED_LENGTH) OR (SELF\IfcStructuralActivity.GlobalOrLocal = GLOBAL_COORDS))
    ' SuitablePredefinedType : PredefinedType <> IfcStructuralCurveActivityTypeEnum.EQUIDISTANT
End Class
